### PR TITLE
Use SAS tokens to get an identity in E2E tests

### PR DESCRIPTION
**Pull request by gauravIoTEdge**
This PR introduces the use of short-lived SAS tokens to get an identity using the AzureCLI task. The use of shared keys to access the blob storage account is now discouraged, prompting this change.

**Testing notes:**

To test, I disabled shared key access on the storage account, then I ran the E2E check-in tests and confirmed that they're passing (and that the logs are uploaded).

image

- [X] I have read the [contribution guidelines](https://github.com/azure/iotedge#contributing).

### CODE REVIEW

File: builds/e2e/templates/e2e-setup.yaml [hunk 1]
```diff
@@ -85,6 +84,38 @@ steps:
...
+    inputs:
+      azureSubscription: $(az.subscription)
+      scriptType: 'bash'
+      scriptLocation: 'inlineScript'
+      inlineScript: |
+        # Hardcoded resource group and storage account name
+        resource_group="EdgeBuilds"
+        storage_account="e2eteststorageedge"
+
+        # Set container name with current date
```
Comment (damonbarry) [inline builds/e2e/templates/e2e-setup.yaml:99]: Comment is out of sync with the code...there's no date in the container name

File: builds/e2e/templates/e2e-setup.yaml [hunk 2]
```diff
@@ -179,6 +210,7 @@ steps:
         loadGenImage = "$imagePrefix-load-gen:$imageTag";
         relayerImage = "$imagePrefix-relayer:$imageTag";
         networkControllerImage = "$imagePrefix-network-controller:$imageTag";
+        testBlobStoreSas = $TestBlobStoreSas;
```
Comment (damonbarry) [inline builds/e2e/templates/e2e-setup.yaml:213]: Does this work? I'd expect that you either need to surround the pipeline var in parens, e.g., `$(TestBlobStoreSas)`, or you have to use the bash variable, which should be ALL CAPS, e.g., `$TESTBLOBSTORESAS`.

Comment (gauravIoTEdge) [reply builds/e2e/templates/e2e-setup.yaml:213]: Well, the tests pass and I verified logs were uploaded.

**Review by damonbarry — APPROVED** [review]

**Pull request merged**